### PR TITLE
feat(scheduler): Add max elapsed duration for model load/unload

### PR DESCRIPTION
**Pull request by sakoush**
Previously we had a fixed elapsed duration defaulting to 15 minutes for model load retries. For loading large llms (e.g. llama3 70b), this fixed max elapsed time didnt allow for any retries consequently as the size of the model will take the duration of the load towards this limit. 

This change then introduces the following:
- Allowing to define a max elapsed duration for model load (defaults to 2 hours) and unload (defaults to 15 minutes)
- Increase an individual load timeout call to 1 hour.
- Reduced the number of retries to 5 from 10.

In the case of retries, hf can resume downloads in the case one download has not finished completely with a single call.

Note that this change doesnt make these values configurable yet for the user to specify.

Fixes: INFRA-1114 (internal).

### CODE REVIEW

File: scheduler/cmd/agent/main.go [hunk 1]
```diff
@@ -47,6 +47,10 @@ const (
 	maxElapsedTimeReadySubServiceBeforeStart = 15 * time.Minute // 15 mins is the default MaxElapsedTime
 	// period for subservice ready "cron"
 	periodReadySubService = 60 * time.Second
+	// max time to wait for a model server to load a model, including retrues
```
Comment (lc525) [inline scheduler/cmd/agent/main.go:50]: nit s/retrues//retries

File: scheduler/pkg/agent/client_utils.go [hunk 2]
```diff
@@ -112,7 +111,7 @@ func (b *backOffWithMaxCount) Reset() {
 }
 
 func (b *backOffWithMaxCount) NextBackOff() time.Duration {
-	if b.currentCount >= b.maxCount {
+	if b.currentCount >= b.maxCount-1 {
```
Comment (lc525) [inline scheduler/pkg/agent/client_utils.go:114]: Just to confirm: does this now retry once if b.maxCount is one? or is it that we now consider the initial function call as part of the "maxCount"?

**Review by lc525 — APPROVED** [review]
lgtm, agreed that we should probably make this configurable

File: scheduler/pkg/agent/client_utils.go (hunk 2 above)
Comment (lc525) [reply scheduler/pkg/agent/client_utils.go:114]: I guess the question is whether for a config parameter named something like (`max*RetryCount`) one would expect a maximum number of `max*RetryCount` retries, or they would expect that the function runs a total of `max*RetryCount` times.

Comment (sakoush) [reply scheduler/pkg/agent/client_utils.go:114]: I added a ticket to create docs to clarify these semantic differences

**Pull request merged**